package/MDAnalysis/analysis/msd.py

```diff
@@ -383,20 +405,27 @@ def _single_frame(self):
         ]
 
     def _conclude(self):
-        if self.fft:
-            self._conclude_fft()
+        if self.non_linear:
+            self._conclude_non_linear()
         else:
-            self._conclude_simple()
+            if self.fft:
+                self._conclude_fft()
+            else:
+                self._conclude_simple()
 
     def _conclude_simple(self):
         r"""Calculates the MSD via the simple "windowed" algorithm."""
         lagtimes = np.arange(1, self.n_frames)
+        dump_times = self.times
         positions = self._position_array.astype(np.float64)
         for lag in tqdm(lagtimes):
             disp = positions[:-lag, :, :] - positions[lag:, :, :]
             sqdist = np.square(disp).sum(axis=-1)
             self.results.msds_by_particle[lag, :] = np.mean(sqdist, axis=0)
         self.results.timeseries = self.results.msds_by_particle.mean(axis=1)
+        self.results.delta_t_values = np.arange(self.n_frames) * [
+            dump_times[1] - dump_times[0]
+        ]
```

Comment:
We're calculating the lag times as `delta_t * [0, 1, 2, ...]`, right? So then we should not use a 1-element list of `[delta_t]`:
```suggestion
self.results.delta_t_values = np.arange(self.n_frames) * (self.times[1] - self.times[0])
```

----

Note that we should actually check that the assumption is true that we're making, namely that really all the delta_t are the same, something like
```python
np.isclose(np.diff(self.times).mean(), self.times[1] - self.times[0])
```

The `run()` method can take the [`frames=...` kwarg](https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html#MDAnalysis.analysis.base.AnalysisBase.run) instead of start/stop/step and if that's the case, we should select your new code path for un-even lag times.